source/_components/withings.markdown

```diff
@@ -0,0 +1,211 @@
+
+```yaml
+# Example configuration.yaml entry
+withings:
+  client_id: CLIENT_ID
+  client_secret: CLIENT_SECRET
+  profiles:
+    - <name of user profile>
+```
+
+Withings supports multiple profiles per account. Each profile has a person's name to help distinguish who's data you're looking at. While the profile provided here can be arbitrary, it is recommended you use the same name from the Withings profile. This will make it easier to distinguish who's data you're looking at.
+
+### {% linkable_title Authorization %}
+
+On the home assistant web page:
+
+- Goto `Configuraton` > `General`
+  - Click `Check Config` and ensure your config looks good.
+  - After config is correct, click `Restart`.
+- After Home assistant restarts.
+  - Got the integrations page.
+  - Toward the bottom of the page, you will see list item for Withings, click it.
+  - Select the profile you intend to pull data.
+  - Click the link to be taken to the withing site to authorize the request.
+  - Withings will provide you with a list of profiles to choose from (if you have more than one). Choose the profile
+  you selected earlier. Otherwise homeassistant entity names will not match the data they are showing.
+- After you authorize at Withings, the Withings website will redirect your browser back to your callback uri.
+
+If your home assistant instance is not accessible publicly, you can change the url in the browser to match the 
+url of your server.
+
+That should do it. Data will begin synchronizing in 5 minutes and update every 5 minutes.
+
+## {% linkable_title FAQ %}
+
+### How can I get data for more than one profile?
+
+Create another sensor entry with the same client id and secret. Just be sure to change the profile to the name of the person you want to track. Then restart home assistant and authorize Withings for that profile.
+
+## {% linkable_title Advanced Configuration %}
+
+{% configuration %}
+client_id:
+  description: The OAuth client id (get from https://account.withings.com/partner/add_oauth2)
+  required: true
+  type: string
+client_secret:
+  description: The OAuth secret (get from https://account.withings.com/partner/add_oauth2)
+  required: true
+  type: string
+profiles:
+  description: Withings supports multiple profiles per account. Provide the person's name who you want home assistant entities to will be associated with (just a name, it doesn't have to be perfect). During the authorization step, you will be asked to select this user from the Withings website.
+  required: true
+  type: map
+base_url:
+  description: The base url to use when authorizing with Withings. This should be the same as the Callback Uri on your Withings account. `<baseurl>/api/withings/callback`
+  required: false
+  type: string
+  default: The base url provided in the `api` component.
+measurements:
+  description: A list of measurements to create sensors for (and track).
+  required: false
+  type: map
+  default: All of the measurements.
+  keys:
```

Comment:
missing a lot of descriptions

Reply:
Are the descriptions necessary? The names are pretty self-explanatory.

Reply:
Yes, they are required.